tests/conftest.py

```diff
@@ -203,7 +205,13 @@ def application_process(
 
 @fixture(scope="session")
 def docker_client() -> DockerClient:
-    return docker.from_env()
+    tests_id = "".join(random.choice(string.ascii_uppercase) for _ in range(10))
```

Comment:
Now it looks good!

Now, 2 final changes I'd like - 
1. Before removing the containers, please note down if any exist. If so, raise an exception AFTER removing them.
2. Let's make the fixture `scope="function"`, this will cause the fixture to error tests which fail to remove their containers, and error them only (unlike `scope="session"` which will mark the session as failed, I bet).

Reply:
wouldn't this fail always in that case - because the exited containers would exist before pruning them and that is the reason we prune them?